llama_index/async_utils.py

```diff
@@ -1,13 +1,39 @@
 """Async utils."""
 import asyncio
 from typing import Any, Coroutine, List
+import nest_asyncio
 
 
-def run_async_tasks(tasks: List[Coroutine]) -> List[Any]:
+def run_async_tasks(
+    tasks: List[Coroutine],
+    show_progress: bool = False,
+    progress_bar_desc: str = "Running async tasks",
+) -> List[Any]:
     """Run a list of async tasks."""
 
+    # jupyter notebooks already have an event loop running
+    # we need to reuse it instead of creating a new one
+    nest_asyncio.apply()
```

Comment:
nit: could we only import `nest_asyncio` do this if we are in notebook environemnt (i.e. have an event loop running already)?

Reply:
yeah in our notebooks we have users manually run this command, no need to put here

Reply:
Jut to clarify this point, should we update the documentation to tell people to run `nest_asyncio.apply()` when using Jupyter? I don't see that right now in the docs.

In general though, where's the best place to education people about this new feature in the docs? (aka "set `show_progress` to `True` if you want...etc.")

Reply:
RE best place to education people in the docs: 
* we can briefly mention it in the top-level usage guide:  https://gpt-index.readthedocs.io/en/latest/guides/primer/usage_pattern.html, since the main configuration surface is on the index class. 
* also the index specific usage guide: https://gpt-index.readthedocs.io/en/latest/how_to/index/usage_pattern.html